packages/router/src/components/prompt/prompt.tsx

```diff
@@ -30,20 +30,18 @@ export class StencilRouterPrompt implements ComponentInterface {
   }
 
   componentWillLoad() {
-    if (this.when ) {
+    if (this.when) {
       this.enable(this.message);
     }
   }
 
   @Watch('message')
   @Watch('when')
   updateMessage(newMessage: string, prevMessage: string) {
-    if (this.when) {
-      if (!this.when || prevMessage !== newMessage) {
-        this.enable(this.message);
-      }
-    } else {
+    if (!this.when) {
       this.disable();
+    } else if (prevMessage !== newMessage) {
+      this.enable(newMessage);
```

Comment:
The changes in this method should just be clean up—doing away with the nested `if` and the redundant `!this.when`